helm-shell.el

```diff
@@ -133,10 +137,10 @@ See `helm-shell-prompts-list'."
 (defun helm-shell-prompts ()
   "Pre-configured `helm' to browse the prompts of the current Shell."
   (interactive)
-  (if (eq major-mode 'shell-mode)
+  (if (cl-member-if 'derived-mode-p helm-shell-comint-mode-list)
```

Comment:
You can use `(apply 'derived-mode-p helm-shell-comint-mode-list)`, which is simpler, same below .